tests/core/full_node/test_full_node.py

```diff
@@ -165,7 +168,12 @@ async def check_transaction_confirmed(transaction) -> bool:
         # Confirm generator is not compressed
         program: Optional[SerializedProgram] = (await full_node_1.get_all_full_blocks())[-1].transactions_generator
         assert program is not None
-        assert detect_potential_template_generator(uint32(5), program) is not None
+        if consensus_mode == ConsensusMode.HARD_FORK_2_0:
+            # after the hard fork we don't use this compression mechanism
+            # anymore, we use CLVM backrefs in the encoding instead
+            assert detect_potential_template_generator(uint32(5), program) is None
+        else:
+            assert detect_potential_template_generator(uint32(5), program) is not None
```

Comment:
I think that here and below it would be worth not copying the value and instead assigning it to a variable that is then used in the two assert conditions.

Reply:
Continued at https://github.com/Chia-Network/chia-blockchain/pull/15977#discussion_r1318547690